barretenberg/cpp/src/barretenberg/stdlib/merkle_tree/membership.hpp

```diff
@@ -315,3 +317,39 @@ void batch_update_membership(field_t<Builder> const& new_root,
 }
 
 } // namespace bb::stdlib::merkle_tree
+
+namespace bb::stdlib {
+/**
+ * @brief Generate a simple merkle tree membership circuit for testing purposes
+ *
+ * @tparam Builder
+ * @param builder
+ * @param num_iterations number of membership checks to perform
+ */
+template <typename Builder> static void generate_merkle_membership_test_circuit(Builder& builder, size_t num_iterations)
```

Comment:
I'd prefer if mock circuits went into their own header files

Reply:
I suppose though if many have them and we're consistent its OK

Reply:
This was a specific request from @codygunton